lib/rspec/core/example_group.rb

```diff
@@ -827,6 +827,12 @@ def self.constant_scope_for(group)
       const_scope
     end
 
+    def self.remove_all_constants
+      self.constants.each do |constant|
+        self.send(:remove_const, constant)
+      end
```

Comment:
We use rubocop to enforce consistent code style, and it's complaining about your use of `self` here:

```
Offenses:
lib/rspec/core/example_group.rb:831:7: C: Redundant self detected.
      self.constants.each do |constant|
      ^^^^^^^^^^^^^^
lib/rspec/core/example_group.rb:832:9: C: Redundant self detected.
        self.send(:remove_const, constant)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

You can see this failure for yourself locally by running `bundle exec rubocop lib`.

The fix is to remove the `self.` as it is unnecessary, and just do `constants.each` and `send` w/o an explicit receiver.

Also, while you're editing this, do you mind changing `send` to `__send__`?  For some slightly obscure compatibility reasons we have a preference for using `__send__` in RSpec's source code over `send`.